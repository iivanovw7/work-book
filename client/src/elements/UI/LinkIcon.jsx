import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { btnBackground, btnLighten } from '../../theme';
/* eslint react/require-default-props: 0 */

const StyledLinkIcon = styled.a`
  display: flex;
  flex-wrap: nowrap;
  padding: 1rem 1rem 1rem 1rem;
  text-align: center;
  flex-direction: row;
  color: ${btnBackground};
  align-items: center;
  text-decoration: none;
  
  &:hover {
    color: ${btnLighten};
    transition: all 0.2s ease-in-out;
    user-select: none;
    cursor: pointer;
  }
  
  span:first-of-type {
  	margin-left: 0.6em;
  }
`;

const LinkIcon = (props) => {
	const {
		link, text, icon, variant, handleClick, theme
	} = props;

	return (
		<ThemeProvider theme={{ mode: theme }}>
			<StyledLinkIcon variant={variant} href={link} onClick={handleClick}>
				<i className="material-icons">
					{icon}
				</i>
				<span>
					{text}
				</span>
			</StyledLinkIcon>
		</ThemeProvider>
	);
};

StyledLinkIcon.propTypes = {
	variant: PropTypes.oneOf(['primary', 'secondary'])
};

StyledLinkIcon.defaultProps = {
	variant: 'primary'
};

LinkIcon.propTypes = {
	text: PropTypes.string.isRequired,
	icon: PropTypes.string.isRequired,
	link: PropTypes.string.isRequired,
	variant: PropTypes.string.isRequired,
	theme: PropTypes.string.isRequired,
	handleClick: PropTypes.func
};

export default LinkIcon;
