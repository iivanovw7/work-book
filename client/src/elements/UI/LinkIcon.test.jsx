import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import LinkIcon from './LinkIcon';

jest.mock('../../config/apiURL', () => () => '/api');

describe('Testing LinkIcon component', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    text: 'text',
    theme: 'dark',
    variant: 'primary',
    icon: 'icon',
    link: 'link'
  };

  const Composition = () => (
    <LinkIcon userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition onClick={mockFunc()} />);

    expect(component).toMatchSnapshot();
  });

  it('Should render with all child elements', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition onClick={mockFunc()} />);

    expect(component.find('i')).toHaveLength(1);
    expect(component.find('span')).toHaveLength(1);

    expect(component.find('i').text()).toBe('icon');
    expect(component.find('span').text()).toBe('text');
  });

  it('Should render link with correct props', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition onClick={mockFunc()} />);

    expect(component.find(LinkIcon).prop('link')).toBe('link');
  });

  it('Should trigger link click event', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition onClick={mockFunc()} />);

    component.find(LinkIcon).at(0).simulate('click');
    expect(component.find(LinkIcon)).toHaveLength(1);
    expect(mockFunc).toHaveBeenCalled();
  });
});
