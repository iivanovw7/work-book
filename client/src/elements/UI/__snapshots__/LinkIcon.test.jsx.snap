// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LinkIcon component Should match snapshot 1`] = `ReactWrapper {}`;
